so.notion.Notion.appdata.xml

```diff
@@ -0,0 +1,40 @@
+<?xml version="1.0" encoding="utf-8"?>
+<component type="desktop-application">
+  <id>so.notion.Notion</id>
+  <metadata_license>CC-BY-SA-3.0</metadata_license>
+  <project_license>LicenseRef-proprietary</project_license>
```

Comment:
The problem I see here is that we don't have rights to redistribution (unless Notion's EULA allows it or you have contacted them and they said it's ok).

It would need to be reworked to use extra-data and run the steps at https://github.com/flathub/flathub/pull/1989/files#diff-8916716d83ed8a4e8d742b7f20df159b50fd930948e3e2877aeb74f3c579147eR116 in apply_extra.

Reply:
I've been trying to make it work with extra-data and apply_extra for a while now, but it's a hot steaming mess - mostly because some of the npm stuff requires `g++` to be available, the patch requires the `patch` tool, the npm cache needs to be preserved inside the image.. just preparing node_modules during the build however also doesn't work because there's an internal repository referenced inside Notion's package.json.

Hence I've reached out to the company and asked for permission. If they decline, I'll try to make it work somehow.

Reply:
Good news, they replied positively, saying that while they won't endorse Linux versions of Notion in the meantime, _the current terms don't disallow this_ and they are _fine ... to try it out_. Would you like to continue the review based on my last pushed commits?

Reply:
OK I got another email, they revoked their initial statement and are not _able to provide written permission to redistribute the program, as this creates potential issues with how we maintain, support, and troubleshoot the app_. I'm not sure I still want to continue at this point after that kind of unsupportive exchange. I'll try a bit more with the extra-data approach and if I can't make it work, close the request.